### PR TITLE
[tools] update TypeDoc and TS, regenerate `unversioned` APIs data

### PR DESCRIPTION
**Pull request by Simek**
# Why

Refs #14912

This PR bumps TypeDoc 

# How

* [tools] update TypeDoc to latest version
* [docs] regenerate `unversioned` APIs data (most of the changes are related to the internal TypeDoc changes (`text` has been changed to `shortText` in some cases), but there are some new, currently missing information added like default values in `FaceDetector`, `KeepAwake` and `Haptics` packages
* [cellular] fix TS error introduce in latest package update
  * Screenshot 2021-10-31 at 17 29 29

# Test Plan

The API data validity has been checked by running docs website on `localhost`.

The changes in Cellular package have been tested using NCL app.

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Review by bbarthec — CHANGES_REQUESTED** [review]
I wonder how it's possible that different `expotools` commands produce different set of TypeScript specific errors? 🤔 
We should track down the root cause of this error 👇

File: packages/expo-cellular/src/ExpoCellular.web.ts [hunk 1]
```diff
@@ -18,7 +18,7 @@ export default {
   },
   async getCellularGenerationAsync(): Promise<CellularGeneration> {
     const connection =
-      navigator.connection || navigator.mozConnection || navigator.webkitConnection || null;
+      navigator['connection'] || navigator.mozConnection || navigator.webkitConnection || null;
```
[the remark this answers is not in the document]
Comment (Simek) [reply packages/expo-cellular/src/ExpoCellular.web.ts:21]: Yup, I can confirm that TS mismatch between workspace an `tools` was a problem. Bumping TS to the version you  have suggested fixed the problem. Thank you for the investigation! 🙂

**Comment by Simek** [comment]
@bbarthec Bumping TS fixed the extraction and `cellular` issue, but it introduced error during tools compilation, ehh.

File: tools/tsconfig.json [hunk 2]
```diff
@@ -18,6 +18,7 @@
     "skipLibCheck": true,
     "skipDefaultLibCheck": true,
     "incremental": true,
+    "useUnknownInCatchVariables": false,
```
Comment (bbarthec) [inline tools/tsconfig.json:21]: How many errors are reported without this one turned off? 🤔

**Review by bbarthec — APPROVED** [review]
LGTM 👍

File: tools/tsconfig.json (hunk 2 above)
Comment (Simek) [reply tools/tsconfig.json:21]: @bbarthec https://github.com/expo/expo/runs/4084290325?check_suite_focus=true

Comment (bbarthec) [reply tools/tsconfig.json:21]: Ok, far too many to try and add everywhere explicit `catch (err: any)` 😅

**Pull request merged**